src/application_input.hpp

```diff
@@ -37,7 +37,10 @@ struct ApplicationInput {
 
   // Mesh functions
   std::function<void(ParameterInput *)> InitUserMeshData = nullptr;
-  std::function<void()> MeshUserWorkInLoop = nullptr;
+  std::function<void(Mesh *, ParameterInput *, SimTime &)> PreStepMeshUserWorkInLoop =
```

Comment:
Question: Does it break anything to take the `Mesh` and `ParameterInput` by references?  Although this is inconsistent with the some of the other functions, the should be required to exist (be non-`NULL`) and, I think, a reference would better capture that requirement.

Reply:
Passing by reference wouldn't break anything. Pointers are passed around everywhere mostly out of convenience and to follow the design of Athena++ on which Parthenon is based.

Mesh is created and held as a `std::unique_ptr` I believe, so `pmesh.get()` makes sense as a thing to pass around. But so would `*pmesh`.

Reply:
Would the preference be to leave it consistent (as it is, I think) with the Athena++ style?

Reply:
Yeah, I think the preference would be leave it consistent for now. But it's also not a huge deal either way.